### PR TITLE
Display systems using a component

### PR DESCRIPTION
**Pull request by gregelin**
Add method controls.element.consuming_systems to produce list of systems
consuming (e.g., containing) the element.

Add tab to component library component detail page to display list of systems
containing the component.

Also, always display OSCAL tab in component library for component detail
(rather than conditional on 'enable_experimental_opencontrol' parameter).

Screen Shot 2021-06-15 at 4 41 31 AM

### CODE REVIEW

File: controls/models.py [hunk 1]
```diff
@@ -326,9 +326,23 @@ def assign_baseline_controls(self, user, baselines_key, baseline_name):
...
         """Return on the statements of statement_type produced by this element"""
+
         smts = Statement.objects.filter(producer_element = self, statement_type = statement_type)
         return smts
 
+    def consuming_systems(self):
+        """Return list of systems for which Element is producer_element of statement of type control_implementation"""
+
+        root_element_ids = set([ce['consumer_element'] for ce in Statement.objects.filter(producer_element=self).values('consumer_element').distinct()])
+        systems = [Element.objects.get(pk=root_element_id).system.all()[0] for root_element_id in root_element_ids if root_element_id]
```
Comment (alexanderward) [inline controls/models.py:337]: Change:
systems = Element.objects.filter(pk__in=root_element_ids)

File: controls/models.py [hunk 2]
```diff
@@ -326,9 +326,23 @@ def assign_baseline_controls(self, user, baselines_key, baseline_name):
...
+        # Remove orphaned systems (e.g., systems whose projects have been deleted). See issue https://github.com/GovReady/govready-q/issues/1617
+        systems_with_projects = []
+        for s in systems:
+            if len(s.projects.all()) > 0:
```
Comment (alexanderward) [inline controls/models.py:341]: Change:
if s.projects.exists()

**Review by alexanderward — CHANGES_REQUESTED** [review]

File: controls/models.py (hunk 2 above)
Comment (gregelin) [reply controls/models.py:341]: Changed.

Comment (gregelin) [reply controls/models.py]: Resolved in 380674c26daf3f29980c27b4480c11a091f4176c

File: controls/models.py (hunk 1 above)
Comment (gregelin) [reply controls/models.py:337]: @alexanderward Looks like what I need is:
systems = System.objects.filter(pk__in=Element.objects.filter(pk__in=root_element_ids).values_list('system', flat=True))

Comment (gregelin) [reply controls/models.py]: Resolved in 5ce09bc5

**Review by davidpofo — APPROVED** [review]
lgtm

**Pull request merged**